cpg-language-go/src/main/kotlin/de/fraunhofer/aisec/cpg/frontends/golang/GoLanguage.kt

```diff
@@ -46,21 +43,62 @@ open class GoLanguage : Language<GoLanguageFrontend>(), HasShortCircuitOperators
     override val startCharacter = '['
     override val endCharacter = ']'
 
+    /** See [Documentation](https://pkg.go.dev/builtin). */
     @Transient
     override val simpleTypes =
         mapOf(
+            // https://pkg.go.dev/builtin#any
+            // TODO: Actually, this should be a type alias to interface{}
+            "any" to ObjectType("any", listOf(), false, this),
+            // https://pkg.go.dev/builtin#error
+            // TODO: Actually, this is an interface{ Error() string } type.
+            "error" to ObjectType("error", listOf(), false, this),
+            // https://pkg.go.dev/builtin#bool
+            "bool" to BooleanType("bool", 1, this),
+            // https://pkg.go.dev/builtin#int
+            "int" to IntegerType("int", 32, this, NumericType.Modifier.SIGNED),
+            // // https://pkg.go.dev/builtin#int8
             "int8" to IntegerType("int8", 8, this, NumericType.Modifier.SIGNED),
+            // https://pkg.go.dev/builtin#int16
             "int16" to IntegerType("int16", 16, this, NumericType.Modifier.SIGNED),
+            // https://pkg.go.dev/builtin#int32
             "int32" to IntegerType("int32", 32, this, NumericType.Modifier.SIGNED),
+            // https://pkg.go.dev/builtin#int64
             "int64" to IntegerType("int64", 64, this, NumericType.Modifier.SIGNED),
+            // https://pkg.go.dev/builtin#uint
+            "uint" to IntegerType("uint", 32, this, NumericType.Modifier.UNSIGNED),
+            // https://pkg.go.dev/builtin#uint8
             "uint8" to IntegerType("uint8", 8, this, NumericType.Modifier.UNSIGNED),
+            // https://pkg.go.dev/builtin#uint16
             "uint16" to IntegerType("uint16", 16, this, NumericType.Modifier.UNSIGNED),
+            // https://pkg.go.dev/builtin#uint32
             "uint32" to IntegerType("uint32", 32, this, NumericType.Modifier.UNSIGNED),
+            // https://pkg.go.dev/builtin#uint64
             "uint64" to IntegerType("uint64", 64, this, NumericType.Modifier.UNSIGNED),
+            // https://pkg.go.dev/builtin#uintptr
+            "uintptr" to
+                IntegerType(
+                    "uint64",
+                    null /* probably 64 bits */,
```

Comment:
It is defined as the maximum bit size to support a pointer (which is probably 64 bits "currently"), should we set it to 64 bits and add a comment?

Reply:
I wouldn't make any assumptions on the architecture if we cannot be sure about it or make the "worst case" assumption. Otherwise, if we set it to 64 bits and e.g. run it on a 32 bit architecture, we will run into problems and we won't be able to catch that bug (even if it won't be likely in this case)

Reply:
Is a uint64 downgraded to a 32 bits on 32 bit architectures or simply not compilable?

Reply:
> Is a uint64 downgraded to a 32 bits on 32 bit architectures or simply not compilable?

Oops that's a copy/paste, error. It should be uintptr. And yes it is 32 bits on a 32 bit architecture, so I will keep it as null because we do not know the architecture.

Reply:
uint64 is not the same as a pointer. The available address space is specific to the architecture, OS and a bunch of things whereas the uint64 would be arranged somewhere in memory (but wouldn't fit into a register)